### ISSUE TITLE
Error when frontmatter value contains colon

### ISSUE THREAD

**Issue by qifei9** [beaussan/update-time-on-edit-obsidian#17]
When the original frontmatter is

```yaml
---
key: "a:b"
---
```

After addition of timestamp by the plugin, it will become

```yaml
---
key: "a:
created: 2021-11-30, 17:55
updated: 2021-11-30, 17:55
---
```

**Comment by beaussan** [comment]
Sorry for this bug, will work on a fix

**Comment by DevJake** [comment]
Can report I'm experiencing the same issue 😬

**Comment by MSBack** [comment]
Same here. I imported a literature note from Zotero using the Zotero Integration plugin, giving me the following YAML:

```
aliases: ["The Intergovernmental Panel on Climate Change: Guardian of Climate Science"]
title: "The Intergovernmental Panel on Climate Change: Guardian of Climate Science"
```
and after the plugin inserted date created and modified, it became:

```
aliases: ["The Intergovernmental Panel on Climate Change
title: "The Intergovernmental Panel on Climate Change
```
Thus rendering the YAML invalid. Probably gonna have to disable the plugin for now.

**Comment by vertis** [comment]
This is a fairly serious bug that is leaving me with quite a number of yaml frontmatter sections that are not just invalid, but have lost data. Is there any update on fixing this problem.

**Comment by laurens-vrh** [comment]
I had this problem as well and my pull request #46 seems to fix it for me. I don't know if the project maintainers are still active, but you should be able to fix it yourself by editing the `main.js` file in the plugin's folder. I did it with a simple find and replace of `split(/: /, 2)` with `split(/: /)`. Hope this helps.

**Comment by beaussan** [comment]
Hi, sorry for the delay. This should now be fixed as of 2.0.0. If that's still an issue, feel free to re open a new issue !

**Comment by guideblade** [comment]
Still not fixed in 2.2.0
